xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp

```diff
@@ -1099,9 +1099,11 @@ bool CAESinkALSA::OpenPCMDevice(const std::string &name, const std::string &para
 void CAESinkALSA::EnumerateDevicesEx(AEDeviceInfoList &list, bool force)
 {
 #if defined(HAVE_LIBUDEV)
-  const auto deviceMonitor = CServiceBroker::GetPlatform().GetService<CALSADeviceMonitor>();
-  if (deviceMonitor)
+  if (CServiceBroker::GetPlatform().HasService<CALSADeviceMonitor>())
+  {
+    const auto deviceMonitor = CServiceBroker::GetPlatform().GetService<CALSADeviceMonitor>();
```

Comment:
Is it guaranteed that between the call to `HasService`and the following `GetService` the service does not vanish? I don't think so and I need to be prepared that `GetService`throws, although `HasService`returned true, right? I know, rather theoretical, but still ...